src/ImportDefinitionsBundle/Command/ExportCommand.php

```diff
@@ -87,7 +88,11 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $eventDispatcher = $this->eventDispatcher;
 
         $params = json_decode($input->getOption('params'), true);
-        $definition = $this->repository->find($input->getOption('definition'));
+        try {
+            $definition = $this->repository->find($input->getOption('definition'));
+        } catch (\InvalidArgumentException $e) {
+            $definition = ExportDefinition::getByName($input->getOption('definition'));
```

Comment:
make use of the repository here

Reply:
I tried this. But is impossible in PhpArrayTable.

Reply:
why? it works with the default repository as well.

Reply:
```
        try {
            $definition = $this->repository->find($input->getOption('definition'));
        } catch (\InvalidArgumentException $e) {
            $definition = $this->repository->findOneBy([
                'name' => $input->getOption('definition')
            ]);
        }
```
Error:
`Call to undefined method addConditionParam in class ImportDefinitionsBundle\Model\ExportDefinition\Listing`

Reply:
True, you shouldn't use that.

What I mean is, you can create the Repository class, inherit from the default one and change the configuration in Configuration.php to use the new Repository Class. In there you have access to the className and you can do what you are doing in the model